processor/queuedprocessor/queued_processor.go

```diff
@@ -150,59 +152,47 @@ func (sp *queuedSpanProcessor) Shutdown(context.Context) error {
 func (sp *queuedSpanProcessor) processItemFromQueue(item *queueItem) {
 	startTime := time.Now()
 	err := sp.sender.ConsumeTraces(item.ctx, item.td)
-	processorStatsTags := []tag.Mutator{tag.Insert(processor.TagProcessorNameKey, sp.name)}
 	if err == nil {
 		// Record latency metrics and return
 		sendLatencyMs := int64(time.Since(startTime) / time.Millisecond)
 		inQueueLatencyMs := int64(time.Since(item.queuedTime) / time.Millisecond)
-		stats.RecordWithTags(context.Background(),
-			processorStatsTags,
+		_ = stats.RecordWithTags(item.ctx,
+			sp.statsTags,
 			statSuccessSendOps.M(1),
 			statSendLatencyMs.M(sendLatencyMs),
 			statInQueueLatencyMs.M(inQueueLatencyMs))
 
 		return
 	}
 
-	spanCountStats := processor.NewSpanCountStats(item.td, sp.name)
-	allSpansCount := spanCountStats.GetAllSpansCount()
-
 	// There was an error
+	_ = stats.RecordWithTags(item.ctx, sp.statsTags, statFailedSendOps.M(1))
 
-	// Immediately drop data on permanent errors. In this context permanent
-	// errors indicate some kind of bad pdata.
-	if consumererror.IsPermanent(err) {
-		sp.logger.Warn(
-			"Unrecoverable bad data error",
-			zap.Int("#spans", spanCountStats.GetAllSpansCount()),
-			zap.Error(err))
-
-		processor.RecordsSpanCountMetrics(
-			context.Background(),
-			spanCountStats,
-			processor.StatBadBatchDroppedSpanCount)
+	sp.logger.Warn("Sender failed", zap.Error(err))
```

Comment:
What's a "Sender"? I don't think we use the term anywhere else in the logs or in the docs, this will confuse the user.

Reply:
Also, do we want to log this for non-permanent errors? Meaning every retry print this. Is this desirable? It can be a lot if the destination has problems.

Reply:
Done. Moved logging the error to onItemDropped so only when we drop items.